### PR TITLE
DB-6504: Sample code for kafka, spark streaming using splice adapter

### PR DESCRIPTION
**Pull request by snowspeeder**

### CODE REVIEW

File: kafka-splice-adapter-streaming/REAME.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+4. Keep kafka producer stream ready (kafka-producer/run_prod.sh), it can be used to stream some mock data to splice-adapater app that will use kafka consumer on the same kafka-topic. 
+
+5. If running on kerberized cluster, copy the hbase users keytab to some common dir such as /tmp on all nodes so that app can reference that. Currently, it will require to use hbase/<masterNode> principal and its keytab. It can cab be found from master node  /var/run/cloudera-scm-agent/process/, searching for latest hbase-MASTER dir, hbase.keytab should be present in that dir. (This step will require sudo privelege)
+
+6. Main.java uner src folder contains a sample streaming app that takes the data from kafka stream to the spark stream and then insert the data to splice engine using splice-spark adapter. You will need to get the build artifacts jars and keep it at the common location say /tmp/dependency on all nodes. Also copy spark jars to /tmp/dependency from the installed base (for cloudera from /opt/cloudera/parcels/SPARK2/lib/spark2/jars, make sure spark-streaming jar is also copied)
```
Comment (dgomezferro) [inline kafka-splice-adapter-streaming/REAME.md]: What jars are you keeping in /tmp/dependency? I think this should work without that step

File: kafka-splice-adapter-streaming/spark_submit.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+TargetTable=TEST_TABLE
+TargetSchema=SPLICE
+RSHostName=localhost
+SpliceConnectPort=1527
+UserName=splice
+UserPassword=admin
+SparkDeployMode=cluster
+ResourceManager=stl-colo-srv056
+NameNodes=stl-colo-srv056
```
Comment (dgomezferro) [inline kafka-splice-adapter-streaming/spark_submit.sh]: Some of this variables are not used

File: kafka-splice-adapter-streaming/spark_submit.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+$SPARK_HOME/bin/spark-submit --class com.splice.custom.reader.Main \
+--master yarn \
+--conf "spark.executor.extraClassPath=/etc/hadoop/conf:/etc/hbase/conf:/tmp/dependency/*:/opt/cloudera/parcels/CDH/lib/hbase/*:$COLON_APPJARS" \
+--jars $APPJARS \
+--deploy-mode $SparkDeployMode \
+--principal $KrbPrincipal \
+--keytab $KrbKeytab \
+--driver-java-options  -Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=4002 \
+--driver-memory 4g \
+--conf "spark.yarn.executor.extraJavaOptions=-Dspark.yarn.appMasterEnv.HADOOP_JAAS_DEBUG=true -Dsun.security.krb5.debug=true -Dsun.security.spnego.debug=true" \
```
Comment (dgomezferro) [inline kafka-splice-adapter-streaming/spark_submit.sh]: is this line useful?

File: kafka-splice-adapter-streaming/src/main/java/com/splice/custom/reader/Main.java [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+    }
+
+    private static void doWork(String inTargetTable, String inTargetSchema, String inHostName, String inHostPort, String inUserName, String inUserPassword, String inKbrPrincipal, String inKbrKeytab, JavaPairDStream<String, String> resultRDD, JavaStreamingContext jssc) throws IOException, InterruptedException {
+
+        SparkConf conf = new SparkConf();
+        SparkSession spark = SparkSession.builder().appName("Reader").config(conf).getOrCreate();
+
+        // Create Splice's Spark Session
+        SpliceClient.isClient = true;
```
Comment (dgomezferro) [inline kafka-splice-adapter-streaming/src/main/java/com/splice/custom/reader/Main.java]: This is not needed.

**Comment by snowspeeder** [comment]
@dgomezferro please take a look again, I addressed all earlier comments, thanks!

**Pull request merged**